File: 0791-custom-sort-string/0791-custom-sort-string.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        unordered_map<char, int> sCount;
        for (char c : s) {
            sCount[c]++;
        }
        string result;
        for (char c : order) {
            if (sCount.find(c) != sCount.end()) {
                int count = sCount[c];
                while (count-- > 0) {
                    result += c;
                }
                sCount.erase(c);
            }
        }      
        for (auto& entry : sCount) {
            int count = entry.second;
            while (count-- > 0) {
                result += entry.first;
            }
        }
        
        return result;
    }
};

File: 0791-custom-sort-string/0791-custom-sort-string.java
import java.util.HashMap;

class Solution {
    public String customSortString(String order, String s) {
        HashMap<Character, Integer> sCount = new HashMap<>();
        for (char c : s.toCharArray()) {
            sCount.put(c, sCount.getOrDefault(c, 0) + 1);
        }

        StringBuilder result = new StringBuilder();
        for (char c : order.toCharArray()) {
            if (sCount.containsKey(c)) {
                int count = sCount.get(c);
                while (count-- > 0) {
                    result.append(c);
                }
                sCount.remove(c);
            }
        }

        for (char c : sCount.keySet()) {
            int count = sCount.get(c);
            while (count-- > 0) {
                result.append(c);
            }
        }
        
        return result.toString();
    }
}

      